firebase-installations/src/main/java/com/google/firebase/installations/time/SystemClock.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.time;
+
+/** @hide */
+public class SystemClock implements Clock {
```

Comment:
I like your thought process, but I would prefer us to use a TimeSource that is offered by / available on Android.

Reply:
The TimeSource that is offered by Android https://developer.android.com/reference/java/time/Clock - requires API level 26

We can't use it to due to the minSDK support.

Reply:
What about JodaTime? Just asking.
I am fine with this implementation.

Reply:
Will have to add a dpendency on `joda-time:joda-time:2.9.4` library which will lead to increase in aar size just to fetch system time.

Reply:
sg, ty
please add a comment in code above System.currentTimeMillies, documenting your decisions.